samples/sensor/kx022/src/main.c

```diff
@@ -0,0 +1,428 @@
+
+	if (rc == -EBADMSG) {
+		/* Sample overrun.  Ignore in polled mode. */
+		if (IS_ENABLED(CONFIG_KX022_TRIGGER)) {
+			overrun = "[OVERRUN] ";
+		}
+		rc = 0;
+	}
+	if (rc == 0) {
+		rc = sensor_channel_get(sensor, SENSOR_CHAN_KX022_MOTION, rd_data);
+	}
+	if (rc < 0) {
+		printf("ERROR: Update failed: %d\n", rc);
+	} else {
+		printf("Motion Direction :\t");
+		motion_direction(&rd_data[0]);
+	}
+}
+
+static void tilt_position_display(const struct device *sensor)
+{
+	struct sensor_value rd_data[2];
+	const char *overrun = "";
+	int rc = sensor_sample_fetch_chan(sensor,SENSOR_CHAN_KX022_TILT);
+
+	if (rc == -EBADMSG) {
+		/* Sample overrun.  Ignore in polled mode. */
+		if (IS_ENABLED(CONFIG_KX022_TRIGGER)) {
+			overrun = "[OVERRUN] ";
+		}
+		rc = 0;
+	}
+	if (rc == 0) {
+		rc = sensor_channel_get(sensor, SENSOR_CHAN_KX022_TILT, rd_data);
+	}
+	if (rc < 0) {
+		printf("ERROR: Update failed: %d\n", rc);
+	} else {
+		printf("Previous Position :\t");
+		tilt_position(&rd_data[0]);
+		printf("Current Position :\t");
+		tilt_position(&rd_data[1]);
+	}
+}
+
+#ifdef CONFIG_KX022_TRIGGER
+static void motion_handler(const struct device *dev, const struct sensor_trigger *trig)
+{
+	static unsigned int motion_cnt;
+
+	fetch_and_display(dev);
+	motion_display(dev);
+	if (++motion_cnt > 5) {
+		motion_test_done = 1;
+		motion_cnt = 0;
+	}
+}
+static void tilt_handler(const struct device *dev, const struct sensor_trigger *trig)
+{
+	static unsigned int slope_cont ;
+
+	fetch_and_display(dev);
+	tilt_position_display(dev);
+	if (++slope_cont > 5) {
+		slope_test_done = 1;
+		slope_cont = 0;
+	}
+}
+#endif
+
+static void test_polling_mode(const struct device *dev)
+{
+	int32_t remaining_test_time = MAX_TEST_TIME;
+
+	printf("\n");
+	printf("\t\tAccelerometer: Poling test Start\r\n");
+	do {
+		fetch_and_display(dev);
+		/* wait a while */
+		k_msleep(SLEEPTIME);
+
+		remaining_test_time -= SLEEPTIME;
+	} while (remaining_test_time > 0);
+}
+
+static void test_trigger_mode(const struct device *dev)
+{
+	struct sensor_trigger trig;
+	uint8_t rc;
+
+	trig.type = SENSOR_TRIG_KX022_MOTION;
+
+	rc = sensor_trigger_set(dev, &trig, motion_handler);
+	printf("\n");
+	printf("\t\tAccelerometer: Motion  trigger test Start\r\n");
+
+	while (motion_test_done == 0) {
+		k_sleep(K_MSEC(200));
+	}
+	motion_test_done = 0;
+	printf("\n");
+	printf("\t\tAccelerometer: Motion trigger test finished\r\n");
+
+	rc = kx022_restore_default_trigger_setup(dev, &trig);
+
+	k_sleep(K_MSEC(500));
+
+	trig.type = SENSOR_TRIG_KX022_TILT;
+
+	rc = sensor_trigger_set(dev, &trig, tilt_handler);
+	printf("\n");
+	printf("\t\tAccelerometer: Tilt Position trigger test Start\r\n");
+
+	while (slope_test_done == 0) {
+		k_sleep(K_MSEC(200));
+	}
+	slope_test_done = 0;
+	printf("\n");
+	printf("\t\tAccelerometer: Tilt Position trigger test finished\r\n");
+
+	rc = kx022_restore_default_trigger_setup(dev, &trig);
+}
+
+static void test_runtime_cfg(const struct device *dev)
+{	
+	//int rc;
```

Comment:
you will need these or else it wont be able to compile if any `CONFIG_KX022_*_RUNTIME` is enabled